File: frontend/src/app/shared/service/collaborator/vocab/VocabularyCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {VocabularyDto} from '../../../model/vocab/Vocabulary.model';
import {VocabularyCriteria} from '../../../criteria/vocab/VocabularyCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class VocabularyCollaboratorService extends AbstractService<VocabularyDto, VocabularyCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/vocabulary/';
    }

    public constrcutDto(): VocabularyDto {
        return new VocabularyDto();
    }

    public constrcutCriteria(): VocabularyCriteria {
        return new VocabularyCriteria();
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/vocab/VocabularyCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.vocab;

import java.util.List;
import ma.zs.alc.bean.core.vocab.Vocabulary;
import ma.zs.alc.dao.criteria.core.vocab.VocabularyCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface VocabularyCollaboratorService extends  IService<Vocabulary,VocabularyCriteria>  {

    List<Vocabulary> findBySectionId(Long id);
    int deleteBySectionId(Long id);
    long countBySectionCode(String code);




}
